src/encord_active/lib/db/data/migrations/202305261617_backfill_label_rows.py

```diff
@@ -0,0 +1,161 @@
+from __future__ import annotations
+
+import json
+from datetime import datetime
+from typing import Optional
+
+from PIL import Image
+from prisma.types import DataUnitCreateInput, LabelRowCreateInput
+from tqdm.auto import tqdm
+
+from encord_active.lib.common.data_utils import get_frames_per_second
+from encord_active.lib.db.connection import PrismaConnection
+from encord_active.lib.project.project_file_structure import (
+    DataUnitStructure,
+    ProjectFileStructure,
+)
+
+
+def up(pfs: ProjectFileStructure):
+    # Adds the content missing from the data units and label rows tables when projects with
+    # older versions of Encord Active are handled with versions greater than 0.1.52.
+    label_row_meta = json.loads(pfs.label_row_meta.read_text(encoding="utf-8"))
+    with PrismaConnection(pfs) as conn:
+        fill_label_rows = conn.labelrow.count() == 0
+        fill_data_units = conn.dataunit.count() == 0
+        old_style_data = conn.labelrow.find_first(where={"label_row_json": "missing"}) is not None or fill_label_rows
+        if not (fill_label_rows or fill_data_units or old_style_data):
+            return
+        migration_counter = tqdm(
+            None,
+            desc=f"Migrating Database: {pfs.project_dir}",
+        )
+        with conn.batch_() as batcher:
+            migration_counter.update(1)
+            for label_row in pfs.iter_labels():
```

Comment:
Need to re-add the iterdir fallback (secret_fallback_iter_files), otherwise migrate will fail if the database is completely empty.